docs/development/pigs/pig-026.rst

```diff
@@ -0,0 +1,287 @@
+
+.. code:: python 
+
+  class Parameter():
+
+        _prior = None
+    
+        @property
+        def prior(self):
+            return self._prior
+    
+        @prior.setter
+        def prior(self, value):
+            self._prior = value
+    
+        def prior_stat_sum(self):
+            if self.prior is not None:
+                return self.prior.weight * self.prior(self)
+
+The ``Parameters`` inherit the priors and evaluate the ``prior_stat_sum`` of all the ``Parameters.parameters``:
+
+.. code:: python 
+
+  class Parameters():
+   
+        @property
+        def prior(self):
+              return [par.prior for par in self]
+          
+          def prior_stat_sum(self):
+              parameters_stat_sum = 0
+              for par in self:
+                  if par.prior is not None:
+                      parameters_stat_sum += par.prior_stat_sum()
+              return parameters_stat_sum
+
+
+During the Fit the ``stat_sum`` of the Datasets is evaluated where now the statistics due to the priors set on the parameters is added. 
+
+.. code:: python 
+
+  class Datasets():
+  
+        def stat_sum(self):
+            """Total statistic given the current model parameters."""
+            stat = self.stat_array()
+            
+            if self.mask is not None:
+                stat = stat[self.mask.data]
+            prior_stat_sum = self.models.parameters.prior_stat_sum()
+            return np.sum(stat, dtype=np.float64) + prior_stat_sum
+
+
+A simple method in the ``Models`` class will allow setting priors on multiple parameters simultaneously. 
+
+.. code:: python 
+
+  class Models(Models):
+      
+        def set_prior(self, parameters, priors):
+            for parameter, prior in zip(parameters, priors):
+                parameter.prior = prior
+
+
+
+
+Serialisation 
+------------
+
+The priors are serialised and read and writeable to a yaml file. They are also saved if the associated ``Parameter`` or ``Parameters`` is transformed to a dictionary. For this, the prior subclasses are tagged. The following two example shows the resulting yaml file for a ``Parameter``.
+
+.. code-block:: yaml
+
+
+  {'name': 'testpar',
+   'value': 0.1,
+   'unit': '',
+   'error': 0,
+   'min': nan,
+   'max': nan,
+   'frozen': False,
+   'interp': 'lin',
+   'scale_method': 'scale10',
+   'is_norm': False,
+   'prior': {'tag': 'GaussianPrior',
+            'parameters': [{'name': 'mu', 'value': 0},
+                           {'name': 'sigma', 'value': 0.1}]}}
+
+
+Implementation Outline
+--------------
+
+The PIG implementation will be piece-wise within individual issues in the following order:
+
+1. ``Prior`` base class and subclasses (See `#4620`_)
+2. ``PriorParameter`` and ``PriorParameters`` (See `#4620`_)
+3. ``.prior`` attribute in the ``Parameter``, ``Parameters`` and ``Model`` classes
+4.  Evaluation of the prior in the ``Datasets`` class
+
+
+
+Decision
+========
+
+
```

Comment:
The PIG was discussed and reviewed among Gammapy developers. It has been reviewed by the CC and is now considered accepted.